src/main/kotlin/com/serameracorp/pattern/PatternService.kt

```diff
@@ -0,0 +1,140 @@
+package com.serameracorp.pattern
+
+import com.serameracorp.project.Project
+import com.serameracorp.project.projectFromResultSet
+import io.ktor.http.*
+import io.ktor.server.plugins.*
+import java.sql.ResultSet
+
+class PatternService {
+
+    val patternRepository = PatternRepository()
+    fun createPattern(formParams: Parameters): Int {
+        val nameParam = formParams["name"]
+        val publisherParam = formParams["publisher"]
+        val publishedInParam = formParams["published_in"]
+        val difficultyParam = formParams["difficulty"]
+
+        patternRepository.createPatternStatement.setString(1, nameParam ?: "")
+        patternRepository.createPatternStatement.setString(2, publisherParam ?: "")
+        patternRepository.createPatternStatement.setString(3, difficultyParam ?: "")
+        patternRepository.createPatternStatement.setString(4, publishedInParam ?: "")
+        val resultSet = patternRepository.createPatternStatement.executeQuery()
+        resultSet.next()
+        return resultSet.getInt("id")
+    }
+
+    fun createPatternFabrics(patternId: Int, formParams: Parameters) {
+        for (x in 0..2) {
+            val fabricParam = formParams["fabric$x"]
+            val fabricLengthParam = formParams["fabric${x}_length"]
+            createPatternFabric(patternId, fabricParam, fabricLengthParam)
+        }
```

Comment:
Similar as above, I'd expect this to take a List of Fabrics to create rather than a Parameters object.